core/recipe-dsl-kotlin/src/main/kotlin/com/ing/baker/recipe/kotlindsl/KotlinDsl.kt

```diff
@@ -129,13 +131,14 @@ class RecipeBuilder(private val name: String) {
         noinline function: (T1, T2, T3) -> R
     ) {
         val parameters = function.reflect()?.parameters ?: error("Cannot read parameters")
-        val ingredients = listOf(javaTypeOf<T1>(), javaTypeOf<T2>(), javaTypeOf<T3>())
+        val javaTypes = listOf(javaTypeOf<T1>(), javaTypeOf<T2>(), javaTypeOf<T3>())
+        val ingredients = javaTypes
             .zip(parameters)
             .map { (clazz, param) -> Ingredient(param.name, clazz) }
-        addSieve(name, ingredients, javaTypeOf<R>(), function)
+        addSieve(name, ingredients, javaTypeOf<R>(), function, javaTypes)
     }
 
-    fun addSieve(name: String, ingredients: List<Ingredient>, returnType: Type, function: Any) {
+    fun addSieve(name: String, ingredients: List<Ingredient>, returnType: Type, function: Any, javaType: List<Type>) {
```

Comment:
javaTypes is a better name here to be more inline with other arguments

Reply:
Very true, fixed it